### PR TITLE
README: Add status badge for upstream daily build

### PR DESCRIPTION
**Pull request by jhedberg**
Add a status badge to indicate whether the upstream daily build is passing or not.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+[![Silabs upstream daily build](https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/workflows/upstream-build.yml/badge.svg)](https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/workflows/upstream-build.yml)
```
Comment (jerome-pouiller) [inline README.md:1]: This is a rather long line. I believe you can get the same result with:

```
[![Silabs upstream daily build][badge]][recipe]

[badge]:  https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/workflows/upstream-build.yml/badge.svg
[recipe]: https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/workflows/upstream-build.yml
```

Comment (jhedberg) [reply README.md:1]: @jerome-pouiller sure, thanks. What I initially had was a copy-paste of what GitHub itself offers when you select "Generate badge" in the Actions tab.

Comment (jhedberg) [reply README.md]: @jerome-pouiller done now, however I trust you that it's actually valid syntax (I'm nota Markdown expert)

Comment (jhedberg) [reply README.md]: Looking at https://github.com/jhedberg/zephyr-silabs/tree/upstream_status_badge it seems to be fine now.

**Review by jerome-pouiller — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jerome-pouiller) [reply README.md]: If someone fork `zephyr-silab` he will get the status of the upstream repo. It is not a big deal I assume.

**Pull request merged**